frontend/iframe/viewmodels/RootViewModel.ts

```diff
@@ -154,6 +173,20 @@ export class RootViewModel extends ViewModel<SegmentType, Options> {
         }
     }
 
+    private async resolveRoomAlias(sessionId: string, roomIdOrAlias: string): Promise<string> {
```

Comment:
`sessionId` isn't actually used in resolving room alias so I am inclined to not have to pass it here.

Reply:
Mmm, it is used to retrieve the `sessionInfo`, which in turn is used to retrieve the `homeserver`, which is then passed to `HomeserverApi`. Right?

Reply:
oops didn't see that. You are right. Now I am wondering if we really need to instantiate a HomeserverAPI instance or we can get away with using `this.platform.config.defaultHomeserver`?

Reply:
`this.platform.config.defaultHomeserver` comes from the block configuration or the popup settings, and is a string, e.g. `matrix.org`. It might be empty, so we can't rely on it being set. Additionally, the user might have changed the homeserver to something else when they logged in.